docs/src/languages/java/migration.md

```diff
@@ -0,0 +1,166 @@
+
+When applications need to access a service, it is important to activate the [Relationships tag](https://docs.platform.sh/configuration/app/relationships.html), because. by default an application may not talk to any other container within a project it includes others projects as a microservices architecture.
+
+After activating the container application to access other services or another application container, the next step is to connect your app to those services. There are several ways to do that, we'll explain the main ones.
+
+### Overwrite
+
+If you are using some frameworks that follow the [Twelve-Factor App](https://12factor.net/), mainly the [third term](https://12factor.net/config) such as Spring, Eclipse MicroProfile Config, Quarkus, Micronauts, and so on, that is easy because you can overwrite the configuration on the application file without add any dependency.
+
+The services information are on the as **PLATFORM_RELATIONSHIPS** [environment that will exist only on a runtime](https://docs.platform.sh/development/variables.html) that is A base64-encoded JSON object whose keys are the relationship name and the values are arrays of relationship endpoint definitions. 
+
+Platform.sh has support to [jq](https://stedolan.github.io/jq/) that allows to extract information from this JSON.
+
+```shell
+export HOST=`echo $PLATFORM_RELATIONSHIPS|base64 -d|jq -r ".database[0].host"`
+```
+
+| Article                                                      | Source                                                       |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Spring Data MongoDB](https://community.platform.sh/t/how-to-overwrite-spring-data-mongodb-variable-to-access-platform-sh-services/528) | [Source](https://github.com/platformsh-examples/java-overwrite-configuration/tree/master/spring-mongodb) |
+| [Jakarta EE/MicroProfile Config](https://community.platform.sh/t/how-to-overwrite-configuration-to-jakarta-microprofile-to-access-platform-sh-services/520) | [Source](https://github.com/platformsh-examples/java-overwrite-configuration/tree/master/jakarta-nosql) |
+| [Spring Data JPA](https://community.platform.sh/t/how-to-overwrite-spring-data-variable-to-access-platform-sh-services/518) | [Source](https://github.com/platformsh-examples/java-overwrite-configuration/tree/master/spring-jpa) |
+| [Payara JPA](https://community.platform.sh/t/how-to-overwrite-variables-to-payara-jpa-access-platform-sh-sql-services/519) | [Source](https://github.com/platformsh-examples/java-overwrite-configuration/blob/master/payara/README.md) |
+
+Several variables might pollute the application files if you wish there is the `.environment` [file as part of your application](https://docs.platform.sh/development/variables.html?#shell-variables). 
```

Comment:
I don't know what this line is saying.  Pollute what, how?

Reply:
Pollute = to many lines in the document.